cli/types/cypress.d.ts

```diff
@@ -2858,7 +2858,7 @@ declare namespace Cypress {
      * Override default config options for E2E Testing runner.
      * @default {}
      */
-    e2e: CoreConfigOptions
+    e2e: Omit<CoreConfigOptions, 'indexHtmlFile'>
```

Comment:
Is there a reason we want this on `CoreConfigOptions` rather than `ComponentConfigOptions`?

Reply:
I think for the `devServer` we're generally going to be accessing the properties off of the un-merged config anyway, because the merging happens in the parent process and we can't serialize/unserialize a user's webpackConfig over the wire (ran into that when refactoring the object API)

> The options in ComponentConfigOptions aren't typed at the root level,

Might not be sure what you're referring to - I was just saying to add the type to this interface:

```
interface ComponentConfigOptions<ComponentDevServerOpts = any> extends Omit<CoreConfigOptions, 'baseUrl'> {
    devServer: DevServerFn<ComponentDevServerOpts>
    devServerConfig?: ComponentDevServerOpts
}
```

And then use it from that to pass into `startServer` directly